src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,40 @@
+interface Props {
+  title: string;
+  name: string;
+  value?: string;
+  label: string;
+  placeholder: string;
+  hasError?: boolean;
+  required?: boolean;
+  onChange: (newValue: string) => void;
+}
+
+export const TextField: React.FC<Props> = ({
+  title,
+  name,
+  value = title,
+  label,
+  placeholder,
+  hasError = false,
+  required = true,
+  onChange,
+}) => {
+  return (
+    <div className="field">
+      <label className="label" htmlFor="title">
+        {label}
+      </label>
+      <input
+        id={name}
+        type="text"
+        data-cy="titleInput"
+        placeholder={placeholder}
+        value={value}
+        onChange={event => onChange(event.target.value)}
+      />
+      {hasError && required && (
+        <span className="error">Please enter a title</span>
```

Comment:
[CODE KNOWLEDGE] - The error message 'Please enter a title' is hardcoded, which might not be appropriate for all instances of the TextField component. Consider using a more generic message or making the error message customizable through props.